qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -139,16 +144,21 @@ def run(self, dag):
                             target_params, target_dag))
 
                 if node.op.params:
-                    # Convert target to circ and back to assign_parameters, since
-                    # DAGCircuits won't have a ParameterTable.
-                    from qiskit.converters import dag_to_circuit, circuit_to_dag
-                    target_circuit = dag_to_circuit(target_dag)
-
-                    target_circuit.assign_parameters(
-                        dict(zip_longest(target_params, node.op.params)),
-                        inplace=True)
-
-                    bound_target_dag = circuit_to_dag(target_circuit)
+                    key = (node.name, tuple(node.op.params))
```

Comment:
Maybe we should have a comment here about the key and what we're looking up. It took me a second to figure out how having this be the cache key would work (because I saw node name and was thinking it was just referring to the pre-bound dag and was worried about returning an incorrect dag from the cache).

Reply:
An unrelated thing here is I think maybe we need a similar property lookup as to DAGNode.condtion so for op nodes we just pull the name directly from the op node. something like:
```python3
@property
def name(self):
   if self._name:
       return self._name
   return self.op.name
```
just because it seems like there is a lot of mixing of node.name and node.op.name and while they should always be the same thing, it's probably good to actually enforce that.

Reply:
Agree this should be consistent (and duplicated state should be removed from the `DAGNode` object). #6110 followed a similar step for `DAGNode.condition`. I'll open an issue.